spec/rspec/expectations/extensions/kernel_spec.rb

```diff
@@ -43,6 +43,16 @@ def method_missing(name, *args)
     it 'works properly on BasicObject-subclassed proxy objects' do
       expect(proxy_class.new(Object.new)).to be_proxied
     end
+
+    it 'does not break the deprecation check on BasicObject-subclassed proxy objects' do
+      begin
+        should_enabled = RSpec::Expectations::Syntax.should_enabled?
+        RSpec::Expectations::Syntax.enable_should unless should_enabled
```

Comment:
This call is idempotent so you don't have to check that it's not enabled before calling it